pkg/loop/internal/test/test_plugin.go

```diff
@@ -0,0 +1,79 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/utils/tests"
+)
+
+type HelperProcessOptions struct {
+	Limit int
+}
+
+func HelperProcess(commandLocation string, command string, opts HelperProcessOptions) *exec.Cmd {
+	cmdArgs := []string{
+		"go", "run", commandLocation, fmt.Sprintf("-cmd=%s", command),
+	}
+	if opts.Limit != 0 {
+		cmdArgs = append(cmdArgs, fmt.Sprintf("-limit=%d", opts.Limit))
+	}
+	cmd := exec.Command(cmdArgs[0], cmdArgs[1:]...) // #nosec
+	cmd.Env = os.Environ()
+	return cmd
+}
+
+func PluginTest[I any](t *testing.T, name string, p plugin.Plugin, testFn func(*testing.T, I)) {
+	ctx, cancel := context.WithCancel(tests.Context(t))
+	defer cancel()
+
+	ch := make(chan *plugin.ReattachConfig, 1)
+	closeCh := make(chan struct{})
+	go plugin.Serve(&plugin.ServeConfig{
+		Test: &plugin.ServeTestConfig{
+			Context:          ctx,
+			ReattachConfigCh: ch,
+			CloseCh:          closeCh,
+		},
+		GRPCServer: plugin.DefaultGRPCServer,
+		Plugins:    map[string]plugin.Plugin{name: p},
+	})
+
+	// We should get a config
+	var config *plugin.ReattachConfig
+	select {
+	case config = <-ch:
+	case <-time.After(5 * time.Second):
```

Comment:
is 5 special? are we setting up for flakey tests by using a hardcoded value?

Reply:
I don't think so: this is just a timeout value. (Note: I just moved this code so would prefer to leave any changes to this out of scope).